### PR TITLE
Add tests for NaT when performing dt.to_period

### LINKED ISSUE #21205: BUG: Series.dt.to_period returns wrong dtype on all-NaTs
#### Code Sample, a copy-pastable example if possible

Input:

```python
print(pd.to_datetime(['2001']).to_series().dt.to_period('D').dtype)
print(pd.to_datetime(['NaT']).to_series().dt.to_period('D').dtype)
```

Output:

```python
object
datetime64[ns]
```

#### Problem description

Calling `.dt.to_period` on an all-NaT series will return datetimes, not periods. This can easily cause programs to crash on unexpected inputs.

#### Expected Output

```python
object
object
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Linux
OS-release: 4.14.14-200.fc26.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: C
LANG: C
LOCALE: None.None

pandas: 0.23.0
pytest: 3.5.1
pip: 10.0.1
setuptools: 39.1.0
Cython: 0.28.2
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 4.2.1
sphinx: 1.7.4
patsy: 0.5.0
dateutil: 2.7.2
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: 3.4.3
numexpr: 2.6.2
feather: None
matplotlib: 2.2.2
openpyxl: None
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.7
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: 0.1.5
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by alexander-ponomaroff**
@mroeschke Here is a test that tests the functionality discussed in the issue. Please let me know if it needs to be done differently and if more tests should be added.

- [x] closes #21205 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/series/test_period.py [hunk 1]
```diff
@@ -164,3 +164,10 @@ def test_end_time_timevalues(self, input_vals):
         result = s.dt.end_time
         expected = s.apply(lambda x: x.end_time)
         tm.assert_series_equal(result, expected)
+
+    def test_to_period(self):
+        series = pd.Series(['2001'], dtype='datetime64[ns]')
```
Comment (mroeschke) [inline pandas/tests/series/test_period.py]: You can parameterize over `'2001', 'NaT'`

File: pandas/tests/series/test_period.py [hunk 2]
```diff
@@ -164,3 +164,10 @@ def test_end_time_timevalues(self, input_vals):
...
+        assert series.dt.to_period('D').dtype == 'Period[D]'
```
Comment (mroeschke) [inline pandas/tests/series/test_period.py]: Construct the expected series and compare using `tm.assert_series.equals(result, expected)` so we can test that more than the dtype is correct.

**Comment by alexander-ponomaroff** [comment]
@mroeschke Thanks for the feedback. Do my changes reflect what you meant?

File: pandas/tests/series/test_period.py [hunk 3]
```diff
@@ -164,3 +164,13 @@ def test_end_time_timevalues(self, input_vals):
...
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('input_vals, expected', [
+        (Series(['2001'], dtype='datetime64[ns]'),
+         Series(['2001'], dtype='period[D]')),
+        (Series(['NaT'], dtype='datetime64[ns]'),
+         Series(['NaT'], dtype='period[D]'))
+    ])
+    def test_to_period(self, input_vals, expected):
+        result = input_vals.dt.to_period('D')
```
Comment (mroeschke) [inline pandas/tests/series/test_period.py]: Could you add the issue number as a comment?

File: pandas/tests/series/test_period.py [hunk 4]
```diff
@@ -164,3 +164,13 @@ def test_end_time_timevalues(self, input_vals):
         result = s.dt.end_time
         expected = s.apply(lambda x: x.end_time)
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('input_vals, expected', [
+        (Series(['2001'], dtype='datetime64[ns]'),
+         Series(['2001'], dtype='period[D]')),
+        (Series(['NaT'], dtype='datetime64[ns]'),
+         Series(['NaT'], dtype='period[D]'))
```
Comment (mroeschke) [inline pandas/tests/series/test_period.py]: Instead, just parameterize over `['2001'. 'NaT']`, then in the test body you can do:

```
expected = Series([arg], dtype='period[D]')
result = Series([arg], dtype='datetime64[ns]').dt.to_period('D')
```

**Review by mroeschke — CHANGES_REQUESTED** [review]
Looks pretty good. A few notes:

File: pandas/tests/series/test_period.py (hunk 4 above)
Comment (alexander-ponomaroff) [reply pandas/tests/series/test_period.py]: That makes a lot more sense. I was making the update late at night and wasn't thinking clearly. Thank you!

**Comment by alexander-ponomaroff** [comment]
@mroeschke Suggestions have been implemented. Thank you.

**Review by mroeschke — APPROVED** [review]
LGTM

**Comment by jreback** [comment]
thanks @alexander-ponomaroff

**Pull request merged**